Automattic/woocommerce-payments
# refactor: express checkout initialization page checks

Fixes #

#### Changes proposed in this Pull Request









Keeping consistent the retrieval of `window.wcpayExpressCheckoutParams`. This refactor opportunity was already brought up during the review process of the tokenized cart for the PRBs.

Instead of accessing `window.wcpayExpressCheckoutParams` directly, we'll just use the `getExpressCheckoutData()` utility.

Since the `is_pay_for_order`/`is_checkout_page`/`is_product_page` provide the same information as `button_context`, I also removed all instances of such booleans in favor of `button_context`.

#### Testing instructions





There shouldn't be a difference in functionality, here.
Since this is affecting the shortcode-based cart/checkout and the product pages, you could check if they're still working.

- Ensure you have the "Enable Cart-Token implementation for ECE" feature flag disabled in the Dev Tools
- As a merchant, ensure you have GooglePay/ApplePay enabled in the WooPayments settings (for product pages, cart, checkout)
- As a customer, go to a product page for a simple product
- You should see the GooglePay button in Chrome
- You can click on the button and you should be able to interact with it
- Close the dialog, add the product to the cart
- Go to the shortcode cart page, you should be able to see the interact with the GooglePay button
- Go to the shortcode checkout page, you should be able to see the interact with the GooglePay button

-------------------


**Post merge**
